### PR TITLE
fix: remove okhttp dependency

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

**Review by csviri — APPROVED** [review]
Just mad one comment. Also not sure if the main pom.xml should be reformatted, but otherwise LGTM

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -50,6 +50,8 @@ jobs:
       - name: Run integration tests in local mode
         working-directory: ${{ matrix.sample_dir }}
         run: |
+          ls /home/runner/.m2/repository/io/fabric8/kubernetes-httpclient-*
+          mvn dependency:tree
```
[the remark this answers is not in the document]
Comment (metacosm) [reply .github/workflows/e2e-test.yml:54]: ah no, we can remove this now, just wanted to check what the hell was going on with the dependencies at some point ;)

File: operator-framework/pom.xml [hunk 2]
```diff
@@ -17,7 +17,11 @@
       <groupId>io.javaoperatorsdk</groupId>
       <artifactId>operator-framework-core</artifactId>
     </dependency>
-
+    <dependency>
+      <groupId>io.fabric8</groupId>
+      <artifactId>kubernetes-httpclient-jdk</artifactId>
```
Comment (csviri) [inline operator-framework/pom.xml:22]: Should not be this the vertx client?

Comment (metacosm) [reply operator-framework/pom.xml:22]: As mentioned, I would like to have tests with different clients to see if there is any impact

File: caffeine-bounded-cache-support/pom.xml [hunk 3]
```diff
@@ -54,6 +54,11 @@
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>io.fabric8</groupId>
+      <artifactId>kubernetes-httpclient-okhttp</artifactId>
```
Comment (csviri) [inline caffeine-bounded-cache-support/pom.xml:59]: Why is okhttp in this module? Probably should be vertx

File: sample-operators/tomcat-operator/pom.xml [hunk 4]
```diff
@@ -38,6 +38,10 @@
             <groupId>io.javaoperatorsdk</groupId>
             <artifactId>operator-framework</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>kubernetes-httpclient-okhttp</artifactId>
```
Comment (csviri) [inline sample-operators/tomcat-operator/pom.xml:43]: Should not be just okhttp in one of these?

File: caffeine-bounded-cache-support/pom.xml (hunk 3 above)
Comment (metacosm) [reply caffeine-bounded-cache-support/pom.xml:59]: Let's test with all the different clients in integration tests in a different PR. In this PR, I wanted to have "some" coverage of the different http client implementations without getting into all possible combinations.

**Review by csviri — APPROVED** [review]

**Pull request merged**